quipucords/api/scanjob/serializer.py

```diff
@@ -90,24 +113,26 @@ class Meta:
         """Metadata for serializer."""
 
         model = ScanJob
-        fields = ['id', 'sources', 'scan_type', 'status', 'status_message',
-                  'tasks', 'options', 'fact_collection_id', 'start_time',
+        fields = ['id',
+                  'scan',
+                  'sources',
+                  'scan_type',
+                  'status',
+                  'status_message',
+                  'tasks',
+                  'options',
+                  'fact_collection_id',
+                  'start_time',
                   'end_time']
 
     @transaction.atomic
     def create(self, validated_data):
         """Create a scan job."""
-        options = validated_data.pop('options', None)
-        scanjob = super().create(validated_data)
-        if options:
-            options = ScanJobOptions.objects.create(**options)
-        else:
-            options = ScanJobOptions()
-        options.save()
-        scanjob.options = options
-        scanjob.save()
-
-        return scanjob
+        scan_job = super().create(validated_data)
+
+        # Copy scan config into job to capture config for historic reference
+        copy_scan_info_into_job(scan_job)
```

Comment:
Maybe consider some kind of `post_init()` method on the ScanJob model, instead of this in the serializer?

Reply:
This is now done when a job is queued.  Basically, the scan configuration is captured when the job is ready to run.

Reply:
I pushed a change for this